test.js

```diff
@@ -26,3 +26,11 @@ test('sync with custom shell', t => {
 	t.true('HOME' in env);
 	t.false('' in env);
 });
+
+test('sync with custom shell throws on non-executable', async t => {
+	t.throws(() => m.sync('non-executable'));
+});
+
+test('async with custom shell throws on non-executable', async t => {
+	await t.throws(m('non-executable'));
```

Comment:
No, you only have to `await`/return it if you want the error object. Otherwise AVA handles waiting for the promise to resolve.

```js
import test from 'ava';

test(t => {
	t.throws(new Promise((resolve, reject) => {
		setTimeout(() => reject(new Error('foo')), 500);
	}));
});
```